DefaultStorage has no attribute "url"

# Bug report

## What's wrong

I am seeing a regression after updating from ``django-stubs==4.2.0`` to ``django-stubs==4.2.3``. This PR appears to be the source of the issue: https://github.com/typeddjango/django-stubs/pull/1488

All of the attributes for the `default_storage` django object now raise type errors.

```python
from django.core.files.storage import default_storage

default_storage.url("filename")
...
```

```
error: "DefaultStorage" has no attribute "url"  [attr-defined]
error: "DefaultStorage" has no attribute "listdir"  [attr-defined]
error: "DefaultStorage" has no attribute "get_modified_time"  [attr-defined]
error: "DefaultStorage" has no attribute "delete"  [attr-defined]
...
```

I understand that this object is difficult to annotate because it's lazy loaded from the user's settings, but I would at least expect the attributes and methods on the base `Storage` class to be allowed.

https://docs.djangoproject.com/en/4.2/ref/files/storage/#the-storage-class

## How is that should be

Using the `default_storage` object, as described by the django docs, should not raise type errors.

## System information

- `python` version: 3.10
- `django` version: 3.1.10
- `mypy` version: 1.4.1
- `django-stubs` version: 4.2.3
- `django-stubs-ext` version: 4.2.2

--- a/django-stubs/core/files/storage/__init__.pyi
+++ b/django-stubs/core/files/storage/__init__.pyi
@@ -24,4 +24,5 @@ def get_storage_class(import_path: str | None = ...) -> type[Storage]: ...
 class DefaultStorage(LazyObject): ...
 
 storages: StorageHandler
-default_storage: DefaultStorage
+# default_storage is actually an instance of DefaultStorage, but it proxies through to a Storage
+default_storage: Storage

--- a/scripts/stubtest/allowlist.txt
+++ b/scripts/stubtest/allowlist.txt
@@ -14,6 +14,9 @@ django.contrib.auth.migrations.*
 django.contrib.flatpages.migrations.*
 django.contrib.contenttypes.migrations.*
 
+# default_storage is actually an instance of DefaultStorage, but it proxies through to a Storage
+django.core.files.storage.default_storage
+
 # BaseArchive abstract methods that take no argument, but typed with arguments to match the Archive and TarArchive Implementations
 django.utils.archive.BaseArchive.list
 django.utils.archive.BaseArchive.extract

